cns/restserver/metrics.go

```diff
@@ -57,13 +63,22 @@ var syncHostNCVersionLatency = prometheus.NewHistogramVec(
 	[]string{"ok"},
 )
 
+var ipRequestFailedCount = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "cx_ip_request_failed_count_total",
+		Help: "Count of the number of times that can not assign IPs when customer want to get an IP address",
+	},
+	[]string{AssignIPConfigFailure, PopulateIPConfigFailure, NoIPsAvailableFailure},
+)
+
 func init() {
 	metrics.Registry.MustRegister(
 		httpRequestLatency,
 		ipAssignmentLatency,
 		ipConfigStatusStateTransitionTime,
 		syncHostNCVersionCount,
 		syncHostNCVersionLatency,
+		ipRequestFailedForNoIPsCount,
```

Comment:
this is undeclared, maybe it should have been
```suggestion
		ipRequestFailedCount,
```